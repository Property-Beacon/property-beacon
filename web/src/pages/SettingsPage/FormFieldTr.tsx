import { Label } from '@redwoodjs/forms'
import { FC } from 'react'

interface Props {
  name: string
  label: string
}

const FormFieldTr: FC<Props> = ({ name, label, children }) => (
  <tr>
    <th className="lg:w-48 hidden md:table-cell capitalize">{label}</th>
    <td className="rounded-lg md:rounded-none">
      <div className="form-control">
        <Label
          name={name}
          className="mb-2 capitalize font-semibold text-sm md:hidden"
        >
          {label}
        </Label>
        {children}
      </div>
    </td>
  </tr>
)

export default FormFieldTr
