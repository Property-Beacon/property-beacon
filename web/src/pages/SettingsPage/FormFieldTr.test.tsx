import { Form } from '@redwoodjs/forms'
import { render, screen } from '@redwoodjs/testing'
import FormFieldTr from './FormFieldTr'

describe('FormFieldTr', () => {
  const mockFn = jest.fn()

  it('renders successfully', () => {
    expect(() => {
      render(
        <Form onSubmit={mockFn} data-testid="test-form">
          <table>
            <tbody>
              <FormFieldTr name="firstName" label="First name" />
            </tbody>
          </table>
        </Form>
      )
    }).not.toThrow()

    expect(screen.getByTestId('test-form')).toMatchSnapshot()
    expect(screen.getByTestId('form-field-tr-th')).toHaveTextContent(
      'First name'
    )
    expect(screen.getByTestId('form-field-tr-label')).toHaveTextContent(
      'First name'
    )
  })

  it('renders customized children', () => {
    expect(() => {
      render(
        <Form onSubmit={mockFn}>
          <table>
            <tbody>
              <FormFieldTr name="address" label="Address">
                <div data-testid="custom-children">Custom Children</div>
              </FormFieldTr>
            </tbody>
          </table>
        </Form>
      )
    }).not.toThrow()

    expect(screen.getByTestId('custom-children')).not.toBeUndefined()
  })
})
