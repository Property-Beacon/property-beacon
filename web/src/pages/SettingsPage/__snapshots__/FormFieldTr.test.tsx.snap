// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldTr renders successfully 1`] = `
<form
  data-testid="test-form"
>
  <table>
    <tbody>
      <tr>
        <th
          class="lg:w-48 hidden md:table-cell capitalize"
          data-testid="form-field-tr-th"
        >
          First name
        </th>
        <td
          class="rounded-lg md:rounded-none"
        >
          <div
            class="form-control"
          >
            <label
              class="mb-2 capitalize font-semibold text-sm md:hidden"
              data-testid="form-field-tr-label"
              for="firstName"
              name="firstName"
            >
              First name
            </label>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</form>
`;
